packages/react-scripts/config/https.js

```diff
@@ -0,0 +1,72 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const crypto = require('crypto');
+const paths = require('./paths');
+
+// Ensure the certificate and key provided are valid and if not
+// throw an easy to debug error
+function validateKeyAndCerts({ cert, key, keyFile, crtFile }) {
```

Comment:
@MikeBeaton I tried an invalid cert without the error handler, and this is what I get:
![image](https://user-images.githubusercontent.com/5689874/58417553-0c787900-807d-11e9-9c47-81f9dd63b8c7.png)
I think in this case it will be needed as otherwise `error:0D07209B:asn1 encoding routines:ASN1_get_object:too long` will mean nothing to most people.

I've changed it to handle the error and include the error message as @Volune suggested.
![image](https://user-images.githubusercontent.com/5689874/58417943-4ac26800-807e-11e9-9698-64984d7cb9d0.png)

Example of key with password error:
![image](https://user-images.githubusercontent.com/5689874/58418111-c1f7fc00-807e-11e9-8be1-9862578f462c.png)

In every invalid cert/key config I've tried, it logs the same error as it does when removing the function, but it also includes the line above saying which file was invalid, which should help users know what they have to change.

Do you think this works as a solution?

Reply:
Hi @alexbrazier - that seems great to me. As long as the full error is there; perfect!